src/chexus/validators.py

```diff
@@ -348,11 +348,17 @@ def applies_to(self, node: Dataset | Group) -> bool:
         )
 
     def validate(self, node: Dataset | Group) -> Violation | None:
-        if not np.isin(
-            node.children['event_id'].value,
-            node.parent.children['detector_number'].value,
-        ).all():
-            return Violation(node.name)
+        event_ids = node.children['event_id'].value
+        detector_numbers = node.parent.children['detector_number'].value
+        if not np.isin(event_ids, detector_numbers).all():
+            show_max = 5
+            diff = np.setdiff1d(event_ids, detector_numbers)[:show_max]
+            return Violation(
+                node.name,
+                'event_id:s that are not in detector_number: '
+                f'{",".join(map(str, diff))}'
+                f'{",..." if len(diff) > show_max else ""}',
+            )
```

Comment:
Can we just make use of NumPy's array formatting? I think it shows up to 1000, but abbreviates after that.